statsmodels/tsa/statespace/sarimax.py

```diff
@@ -0,0 +1,1941 @@
+    r"""
+    Seasonal AutoRegressive Integrated Moving Average with eXogenous regressors
+    model
+
+    Parameters
+    ----------
+    endog : array_like
+        The observed time-series process :math:`y`
+    exog : array_like, optional
+        Array of exogenous regressors, shaped nobs x k.
+    order : iterable or iterable of iterables, optional
+        The (p,d,q) order of the model for the number of AR parameters,
+        differences, and MA parameters. `d` must be an integer
+        indicating the integration order of the process, while
+        `p` and `q` may either be an integers indicating the AR and MA
+        orders (so that all lags up to those orders are included) or else
+        iterables giving specific AR and / or MA lags to include. Default is
+        an AR(1) model: (1,0,0).
+    seasonal_order : iterable, optional
+        The (P,D,Q,s) order of the seasonal component of the model for the
+        AR parameters, differences, MA parameters, and number of seasons.
+        `d` must be an integer indicating the integration order of the process,
+        while `p` and `q` may either be an integers indicating the AR and MA
+        orders (so that all lags up to those orders are included) or else
+        iterables giving specific AR and / or MA lags to include. `s` is an
+        integer giving the number of seasons, often it is 4 for quarterly data
+        or 12 for monthly data. Default is no seasonal effect.
+    trend : str{'n','c','t','ct'} or iterable, optional
+        Parameter controlling the deterministic trend polynomial :math:`A(t)`.
+        Can be specified as a string where 'c' indicates a constant (i.e. a
+        degree zero component of the trend polynomial), 't' indicates a
+        linear trend with time, and 'ct' is both. Can also be specified as an
+        iterable defining the polynomial as in `numpy.poly1d`, where
+        `[1,1,0,1]` would denote :math:`a + bt + ct^3`. Default is to not
+        include a trend component.
+    measurement_error : boolean, optional
+        Whether or not to assume the observations were measured with error.
+        Default is False.
+    time_varying_regression : boolean, optional
+        Used when an exogenous dataset is provided to select whether or not
+        coefficients on the exogenous regressors are allowed to vary over time.
+        Default is False.
+    mle_regression : boolean, optional
+        Whether or not to use estimate the regression coefficients as part of
+        maximum likelihood estimation or through the Kalman filter (i.e.
+        recursive least squares). If `time_varying_regression` is True, this
+        must be set to False. Default is True.
+    simple_differencing : boolean, optional
+        Whether or not to use conditional maximum likelihood estimation.
+        If True, differencing is performed prior to estimation, which discards
+        the first :math:`SD + d` initial rows but reuslts in a smaller
```

Comment:
Ah, good point, this is indeed only partially conditional, since after simple differencing (`s D + d`), we can still make use of all remaining data regardless of the AR order.